### PR TITLE
Support direct signaling for advancing advertiser cleanroom state

### PR DESCRIPTION
**Pull request by amanjpro**
This PR makes it possible to directly ping the backend of the Optable DCN to advance the state of the PAIR advertiser.
That way they don't need to wait for the pubsub notification to ping the backend.

This will help with local testing and system testing

### CODE REVIEW

File: pkg/cmd/cli/pair.go [hunk 1]
```diff
@@ -84,6 +84,15 @@ func (c *pairConfig) hashEncryt(ctx context.Context, input string) (err error) {
 	if err != nil {
 		return fmt.Errorf("bucket.NewBucketCompleter: %w", err)
 	}
+	if err != nil {
+		return fmt.Errorf("completionChecker.Check: %w", err)
+	}
```
Comment (mariiatuzovska) [inline pkg/cmd/cli/pair.go:89]: err checked twice

File: pkg/cmd/cli/pair.go [hunk 2]
```diff
@@ -84,6 +84,15 @@ func (c *pairConfig) hashEncryt(ctx context.Context, input string) (err error) {
...
+	hasCompleted, err := bucketCompleter.HasCompleted(ctx)
```
Comment (mariiatuzovska) [inline pkg/cmd/cli/pair.go:90]: should you check err?

**Review by mariiatuzovska — CHANGES_REQUESTED** [review]

File: pkg/cmd/cli/pair.go (hunk 1 above)
Comment (juanli16) [inline pkg/cmd/cli/pair.go:89]: this meant for the call below

File: pkg/cmd/cli/pair.go [hunk 3]
```diff
@@ -251,3 +270,11 @@ func (c *pairConfig) match(ctx context.Context, outputPath string, publisherPAIR
...
 	return nil
 }
+
+func readersFromReadClosers(rs []io.ReadCloser) []io.Reader {
+	readers := make([]io.Reader, len(rs))
+	for i, r := range rs {
+		readers[i] = r
+	}
+	return readers
+}
```
Comment (juanli16) [inline pkg/cmd/cli/pair.go:280]: still needed?

Comment (amanjpro) [reply pkg/cmd/cli/pair.go:280]: It is actually still needed, we still perform matches, but not as a standalone command

Comment (juanli16) [reply pkg/cmd/cli/pair.go:280]: ah this was from the removed files

**Review by juanli16 — APPROVED** [review]

**Review by mariiatuzovska — APPROVED** [review]

**Pull request merged**